File: benchmark/prolog/task3_fizzbuzz.pl
fizzbuzz(N) :-
    ( 0 is N mod 15 -> write('FizzBuzz')
    ; 0 is N mod 3 -> write('Fizz')
    ; 0 is N mod 5 -> write('Buzz')
    ; write(N)
    ), nl.

main :- forall(between(1, 100, N), fizzbuzz(N)).

File: benchmark/ocaml/task3_fizzbuzz.ml
for i = 1 to 100 do
  match (i mod 3, i mod 5) with
  | (0, 0) -> print_endline "FizzBuzz"
  | (0, _) -> print_endline "Fizz"
  | (_, 0) -> print_endline "Buzz"
  | _ -> print_endline (string_of_int i)
done
